### ISSUE TITLE
Docs: run step 'build' on all PRs; run 'deploy' step on merge

### ISSUE THREAD

**Issue by antobinary** [bigbluebutton/bigbluebutton#16974]
We currently only run the steps `build` and `deploy` together, and do it for certain branches;
This means when we open a new PR with docs changes,  we do not know automatically if there is a build-breaking change

Perhaps we can allow

```
on:
    pull_request:
    types: [opened, synchronize, reopened]
```

to run `build` step

and restrict `deploy` step for only after we merge
